### PR TITLE
Error Handling and Test Coverage for CheckoutApiException

### PR DESCRIPTION
**Pull request by armando-rodriguez-cko**
### Description
This PR improves the `CheckoutApiException` class and its associated tests to ensure more robust error handling for scenarios where API responses are partial or do not contain the expected keys (`error_codes` or `error_type`).

### Changes Made
- **Enhanced error handling in `CheckoutApiException`**:
  - Replaced direct dictionary access (`payload['key']`) with `payload.get('key')` to prevent `KeyError` in cases of incomplete JSON responses.
  - Ensured `error_details` and `error_type` attributes default to `None` if the corresponding keys are not present in the response payload.

- **Updated unit tests**:
  - Covered scenarios where API responses:
    - Do not include keys like `error_codes` or `error_type`.
    - Are valid but contain no relevant data.
    - Return empty text or invalid JSON.

- **Improved robustness**:
  - Optimized the implementation to handle partial responses gracefully, ensuring a more consistent experience for SDK users.

### Impact
- Improves the stability and robustness of the SDK in error-handling scenarios.
- Increases unit test coverage for the `CheckoutApiException` class.
- Simplifies error diagnostics by explicitly handling edge cases.

### Additional Notes
- This change does not introduce any breaking changes and remains fully backward compatible with existing implementations.

### CODE REVIEW

File: checkout_sdk/exception.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 class CheckoutException(Exception):
 
     def __init__(self, message=None):
```
Comment (robert-goheen-cko) [inline checkout_sdk/exception.py:9]: I don't know Python that well, but can't you just have the default set to empty string here rather than using the conditional below?

**Review by robert-goheen-cko — APPROVED** [review]

File: checkout_sdk/exception.py (hunk 1 above)
Comment (armando-rodriguez-cko) [reply checkout_sdk/exception.py:9]: None is the Python equivalent of “no value” or “undefined” (similar to null in other languages). Using None allows the SDK to handle optional parameters, deferred initialisation, and missing data gracefully without introducing ambiguous default values.
One important reason for using None in exception classes is to avoid undesired behavior when converting an empty message to a string. If None is passed as the message and directly converted to a string (e.g., str(None)), Python will produce the string "None", which is likely unintended and confusing for users.
